src/GovUk.Education.ExploreEducationStatistics.Content.Model/Database/ContentDbContext.cs

```diff
@@ -535,6 +535,15 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
             modelBuilder.Entity<KeyStatisticDataBlock>()
                 .ToTable("KeyStatisticsDataBlock");
+
+            modelBuilder.Entity<KeyStatisticDataBlock>()
+                .HasOne<DataBlock>(ks => ks.DataBlock)
+                .WithMany()
+                // WARN: This is necessary - otherwise a cascade delete occurs when an associated data block is removed.
+                // This cascade delete _only_ removes the KeyStatisticsDataBlock entry, leaving a KeyStatistics table
+                // entry.
+                .OnDelete(DeleteBehavior.NoAction);
+
```

Comment:
`DataBlockService.DeleteDataBlocks` is still saving changes before the `KeyStatisticsDataBlock` is deleted, so I would have thought right now there's still a bug where that fails when saving changes?

Reply:
Discussed on Slack - this has been done. I also checked for other locations where this might happen. I believe the only place is in `ContentService#RemoveContentBlock`, but that code should never get run and will get removed in the cleanup ticket EES-3988, so I'm content to leave that.